corehq/apps/domain/decorators.py

```diff
@@ -241,6 +241,22 @@ def __inner__(request, domain, *args, **new_kwargs):
     return decorator
 
 
+def api_domain_view(view):
+    """
+    Decorate this with any domain view that should be accessed via api only
+    """
+    @wraps(view)
+    @api_key()
+    @login_and_domain_required
+    def _inner(request, domain, *args, **kwargs):
+        if request.user.is_authenticated():
+            request.couch_user = CouchUser.from_django_user(request.user)
```

Comment:
`login_and_domain_required` doesn't set the `request.couch_user`, that's the reason this decorator is created.